src/main/kotlin/org/opensearch/replication/action/stats/TransportLeaderStatsAction.kt

```diff
@@ -0,0 +1,100 @@
+        TransportNodesAction<LeaderStatsRequest, LeaderStatsResponse, NodeStatsRequest, LeaderNodeStatsResponse>(LeaderStatsAction.NAME,
+             threadPool, clusterService, transportService,  actionFilters, ::LeaderStatsRequest,  ::NodeStatsRequest, ThreadPool.Names.MANAGEMENT,
+                LeaderNodeStatsResponse::class.java), CoroutineScope by GlobalScope {
+
+    companion object {
+        private val log = LogManager.getLogger(TransportLeaderStatsAction::class.java)
+        val durationThresholdActiveReplication =TimeUnit.SECONDS.toMillis(300) //5 min
+    }
+
+    override fun newNodeRequest(request: LeaderStatsRequest): NodeStatsRequest {
+       return NodeStatsRequest()
+    }
+
+    override fun newNodeResponse(input: StreamInput): LeaderNodeStatsResponse {
+        return LeaderNodeStatsResponse(input)
+    }
+
+    override fun newResponse(request: LeaderStatsRequest?, responses: MutableList<LeaderNodeStatsResponse>?, failures: MutableList<FailedNodeException>?): LeaderStatsResponse {
+        return LeaderStatsResponse(clusterService.clusterName, responses, failures)
+
+    }
+
+     private fun isIndexReplicationActive(index :String) :Boolean {
+        val indicesRouting = clusterService.state().routingTable.indicesRouting
+         if (!indicesRouting.containsKey(index)) {
+             return false
+         }
+
+         return true
+         // Retention lease is the most accurate method to find out active replication, however it blocks the generic thread pool
+         // given that nodeOperation doesn't support async calls
+/*
+
+        var leaseExist = false
+
+        var req = client.admin().indices().prepareStats().clear().setStore(true).setIndices(index).request()
+
+         runBlocking  {
+             var indicesStatsResponse = client.suspendExecute(IndicesStatsAction.INSTANCE, req)
+            for (shardStats in indicesStatsResponse.shards) {
+                for (retentionLease in shardStats.retentionLeaseStats.retentionLeases().leases()) {
+                    if (retentionLease.id().startsWith(RETENTION_LEASE_PREFIX)) {
+                        log.info("Lease name  ${retentionLease.id()} $index")
+                        leaseExist = true
+                    }
+                }
+            }
+            log.info("Retention lease exists $leaseExist $index")
+        }
+
+
+         return leaseExist
+
+ */
+    }
+
+    override fun nodeOperation(nodeStatRequest: NodeStatsRequest?): LeaderNodeStatsResponse {
+        var activeStats = remoteStats.stats.filter { (k, v) ->
+            var indexExists = isIndexReplicationActive(k)
+            val timeSinceLastFetch = TimeUnit.NANOSECONDS.toMillis( System.nanoTime() - v.lastFetchTime.get() )
+            (timeSinceLastFetch < durationThresholdActiveReplication ) && indexExists
+        }
+
+       return LeaderNodeStatsResponse(this.clusterService.localNode(), activeStats)
+    }
+}
```

Comment:
Can you ensure all files have correct line endings?

Reply:
Fixed